src/uct/base/uct_iface.c

```diff
@@ -554,38 +554,35 @@ ucs_config_field_t uct_iface_config_table[] = {
 ucs_status_t uct_base_ep_am_short_iov(uct_ep_h ep, uint8_t id, const uct_iov_t *iov,
                                       size_t iovcnt)
 {
-    size_t length = 0;
-    unsigned buffer_length;
+    uint64_t header = 0;
+    size_t length;
     void *buffer;
     size_t iov_it;
-    void *payload;
+    size_t offset;
     ucs_status_t status;
 
-    buffer_length = ucs_max(uct_iov_total_length(iov, iovcnt), sizeof(uint64_t));
-    if (buffer_length > UCS_ALLOCA_MAX_SIZE) {
-        buffer = malloc(buffer_length);
+    length = uct_iov_total_length(iov, iovcnt);
+    if (length > UCS_ALLOCA_MAX_SIZE) {
+        buffer = ucs_malloc(length, "uct_base_ep_am_short_iov buffer");
     } else {
-        buffer = ucs_alloca(buffer_length);
+        buffer = ucs_alloca(length);
     }
-    memset(buffer, 0, sizeof(uint64_t));
 
-    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
-        memcpy(UCS_PTR_BYTE_OFFSET(buffer, length), iov[iov_it].buffer,
+    for (iov_it = 0, offset = 0; iov_it < iovcnt; ++iov_it) {
+        memcpy(UCS_PTR_BYTE_OFFSET(buffer, offset), iov[iov_it].buffer,
                iov[iov_it].length);
-        length += iov[iov_it].length;
+        offset += iov[iov_it].length;
     }
 
-    if (length > sizeof(uint64_t)) {
-        payload = UCS_PTR_BYTE_OFFSET(buffer, sizeof(uint64_t));
-        length  = length - sizeof(uint64_t);
-    } else {
-        payload = NULL;
-        length  = 0;
-    }
+    offset = ucs_min(sizeof(header), length);
```

Comment:
add comment that we send at least 8b? and if user data is smaller, we pad with 0